packages/jellyfish-api-core/__tests__/category/loan/paybackLoan.test.ts

```diff
@@ -532,30 +580,52 @@ describe('paybackLoan success', () => {
+    expect(vaultBefore.interestAmounts).toStrictEqual([`${tslaInterestTotalBefore.toFixed(8)}@TSLA`])
+    expect(vaultBefore.loanValue).toStrictEqual(tslaLoanValueBefore.toNumber())
+    expect(vaultBefore.interestValue).toStrictEqual(tslaInterestValueBefore.toNumber())
+    expect(vaultBefore.collateralRatio).toStrictEqual(18750) // (15000 / tslaLoanValueBefore) * 100
     expect(vaultBefore.informativeRatio).toStrictEqual(18749.98929375)
 
-    const utxo = await bob.container.fundAddress(bobloanAddr, 250)
+    const utxo = await bob.container.fundAddress(bobloanAddr, 250) // one extra block has been generate here and so the interest has been increased here
+
+    const heightAfterFundAddress = await bob.container.getBlockCount()
+    const totalTslaInterestAfterFundAddress = tslaInterestTotalBefore.plus(tslaInterestPerBlockBefore.multipliedBy(new BigNumber(heightAfterFundAddress - height)))
 
+    const tslaPaybackAmount = 13
     const txid = await bob.rpc.loan.paybackLoan({
       vaultId: bobVaultId,
-      amounts: '13@TSLA',
+      amounts: `${tslaPaybackAmount}@TSLA`,
       from: bobloanAddr
-    }, [utxo])
+    })
     expect(typeof txid).toStrictEqual('string')
     await bob.generate(1)
 
+    // check the interests after payback
+    // total interest payable is paid back upfront even at partial payback. Becuase of that, actual loanDecreasedAfterPayback = (tslaPaybackAmount - total interest)
+    const loanDecreasedAfterPayback = new BigNumber(tslaPaybackAmount).minus(totalTslaInterestAfterFundAddress)
+
+    const tslaInterestPerBlockAfter = tslaInterestPerBlockBefore.minus(loanDecreasedAfterPayback.multipliedBy(netInterest).dividedBy(365 * blocksPerDay))
+    const tslaInterestTotalAfter = tslaInterestPerBlockAfter.multipliedBy(1).dp(8, BigNumber.ROUND_CEIL) // total interest is always ceild to CAmount precision in AIN
+    const tslaLoanAmountAfter = new BigNumber(40).minus(loanDecreasedAfterPayback).plus(tslaInterestTotalAfter)// tslaLoanAmountBefore - loanDecreasedAfter + tslaInterestTotalAfter
+
+    const tslaLoanValueAfter = tslaLoanAmountAfter.dp(8, BigNumber.ROUND_FLOOR).multipliedBy(new BigNumber(2)) // loanAmount * 2 (1 TSLA = 2 USD)
+    const tslaInterestValueAfter = tslaInterestTotalAfter.multipliedBy(new BigNumber(2))
+
     const vaultAfter = await bob.container.call('getvault', [bobVaultId])
-    expect(vaultAfter.loanAmounts).toStrictEqual(['27.00006109@TSLA']) // 40.00002283 - 13 + totalInterest
-    expect(vaultAfter.interestAmounts).toStrictEqual(['0.00001541@TSLA'])
-    expect(vaultAfter.loanValue).toStrictEqual(54.00012218) // 27.00006109 * 2 (::1 TSLA = 2 USD)
-    expect(vaultAfter.interestValue).toStrictEqual(0.00003082)
-    expect(vaultAfter.collateralRatio).toStrictEqual(27778) // 15000 / 54.00012218 * 100
+    expect(vaultAfter.loanAmounts).toStrictEqual([`${tslaLoanAmountAfter.toFixed(8, BigNumber.ROUND_FLOOR)}@TSLA`])
```

Comment:
inside ain, only the interest is stored and calculated in 128 bit. loan amounts are still in 64 bit. since these are integer calculations, only the quotient will be taken into account after a division. the remainder is lost, and this is equivalent to the `FLOOR` operation.